code/game/objects/items/potions.dm

```diff
@@ -0,0 +1,191 @@
+/obj/item/potion
+	name = "potion"
+	desc = "This doesn't look like it does anything."
+	icon = 'icons/obj/potions.dmi'
+	icon_state = "red_minibottle"
+	w_class = W_CLASS_SMALL
+	slot_flags = SLOT_BELT
+	var/full = TRUE
+	var/fail_message = "<span class='notice'>Nothing happens, though your stomach is a little unsettled. It seems the potion isn't agreeing with you.</span>"
+
+/obj/item/potion/attack_self(mob/user)
+	if(full)
+		user.visible_message("<span class='danger'>\The [user] drinks \the [src].</span>", "<span class='notice'>You drink \the [src].</span>")
+		playsound(get_turf(src),'sound/items/uncorking.ogg', rand(10,50), 1)
+		spawn(6)
+			playsound(get_turf(src),'sound/items/drink.ogg', rand(10,50), 1)
+		imbibe(user)
+
+/obj/item/potion/update_icon()
+	if(full)
+		icon_state = initial(icon_state)
+	else
+		icon_state = "[copytext("[initial(icon_state)]",findtext("[initial(icon_state)]","_")+1)]"
```

Comment:
Make this a direct variable instead. This is unreadable and messy on so many levels.

Reply:
I could split it up into a couple lines so it's more readable.
Having it precludes the need for a separate variable just to hold the empty state.

Reply:
Fine I guess, but make sure to comment it too.